### PR TITLE
DROOLS-5250 (ex KOGITO-1685): Wrong label name for multiple nested elements

### PR DESCRIPTION
**Pull request by yesamer**
@danielezonca @dupliaka Can you please review and test?

Basically, the label  of nested properties was wrong in the case of multiple nested element. 
This is because, multiple nested properties shared the same list used to determine the label and the `expressionElement` list. 
To solve the issue, I simply created a new list of previous step for any nested property.

The fix is applied for both KOGITO and BC. Considering they are not sharing this part of code (in BC is on server side), the fix is applied in 2 different points

https://issues.redhat.com/browse/DROOLS-5250

### CODE REVIEW

**Comment by dupliaka** [comment]
jenkins execute full downstream build

**Comment by dupliaka** [comment]
Unfortunately I could not find out a variable "nested" in tNested of  Testing Tools
See: 
image
 Please take a look

**Comment by dupliaka** [comment]
For context type after this update i noticed that i can create two facts of the same types:
image

**Comment by dupliaka** [comment]
Types are non connected 
image
tNested does not contains string parameter nested
And contextType contains value but should not.

**Comment by yesamer** [comment]
@dupliaka  I'll manage these issue in https://issues.redhat.com/browse/KOGITO-1747, where I'm fixing several issues (not only the collection one).

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/KogitoScenarioSimulationBuilder.java [hunk 1]
```diff
@@ -359,7 +359,7 @@ protected static double getColumnWidth(String expressionIdentifierName) {
         }
     }
 
-    private void addFactMapping(final FactMappingExtractor factMappingExtractor,
+    protected void addFactMapping(final FactMappingExtractor factMappingExtractor,
                                 final FactModelTree factModelTree,
                                 final List<String> previousSteps,
                                 final Map<String, FactModelTree> hiddenValues) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/KogitoScenarioSimulationBuilder.java:365]: nitpicking - formatting is broken

Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/KogitoScenarioSimulationBuilder.java]: @jomarko Done :)

**Pull request merged**